server/src/test/java/org/elasticsearch/cluster/routing/allocation/decider/IndexBalanceAllocationDeciderTests.java

```diff
@@ -75,25 +72,19 @@ public class IndexBalanceAllocationDeciderTests extends ESAllocationTestCase {
     private IndexMetadata indexMetadata;
     private RoutingAllocation routingAllocation;
     private IndexBalanceAllocationDecider indexBalanceAllocationDecider;
-    private int excessShards;
     private ShardRouting indexTierShardRouting;
     private ShardRouting searchTierShardRouting;
     private List<RoutingNode> indexTier;
     private List<RoutingNode> searchIier;
 
     private void setup(Settings settings) {
-        boolean hasDiscoveryNodeFilters = settings.getAsBoolean(INCLUDE_DISCOVERY_NODE_FILTERS, true);
-        boolean allowExcessShards = settings.getAsBoolean(ALLOW_EXCESS_SHARDS, true);
-
         final String indexName = "IndexBalanceAllocationDeciderIndex";
         final Map<DiscoveryNode, List<ShardRouting>> nodeToShardRoutings = new HashMap<>();
 
-        excessShards = allowExcessShards ? randomIntBetween(1, 5) : 0;
-
         Settings.Builder builder = Settings.builder()
             .put("stateless.enabled", "true")
-            .put(IndexBalanceConstraintSettings.INDEX_BALANCE_DECIDER_ENABLED_SETTING.getKey(), "true")
-            .put(IndexBalanceConstraintSettings.INDEX_BALANCE_DECIDER_EXCESS_SHARDS.getKey(), excessShards);
+            .put(IndexBalanceConstraintSettings.INDEX_BALANCE_DECIDER_ENABLED_SETTING.getKey(), "true");
+        builder.put(settings);
```

Comment:
maybe it's worth putting these first, I assume we don't want to allow callers to override `stateless.enabled` or decider enabled? Or maybe we do.